src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1732,6 +1737,13 @@ private int tryExecute(final Runnable runnable) {
             }
             assert tailNext != tail;
             if (tailNext instanceof TaskNode) {
+                // we trust the producer moving tail won't be descheduled
+                // hence there is no need to help it
+                if (UNDER_SUBSCRIBED) {
+                    tail = this.tail;
```

Comment:
In this case do you think we should increment spin misses and call `onSpinWait` since we may be waiting for another thread to complete an action?

Reply:
I really don't have any idea, maybe yes, but I'm not sure it has the same meaning with threads < cores because we always performs offers in 2 steps and given that we're not allowed to "help" the active producer is a "common" scenario that we can see an operation in an "half-completed" state...Just thinking loud...